src/Files.App/Views/Shells/BaseShellPage.cs

```diff
@@ -248,36 +248,39 @@ protected async void FilesystemViewModel_DirectoryInfoUpdated(object sender, Eve
 					? await GitHelpers.GetRepositoryHead(InstanceViewModel.GitRepositoryPath)
 					: null;
 
-			if (InstanceViewModel.GitRepositoryPath != FilesystemViewModel.GitDirectory)
+			if (FilesystemViewModel.GitDirectory is not null)
 			{
-				InstanceViewModel.GitRepositoryPath = FilesystemViewModel.GitDirectory;
+				if (InstanceViewModel.GitRepositoryPath != FilesystemViewModel.GitDirectory)
+				{
+					InstanceViewModel.GitRepositoryPath = FilesystemViewModel.GitDirectory;
 
-				InstanceViewModel.GitBranchName = headBranch is not null
-					? headBranch.Name
-					: string.Empty;
+					InstanceViewModel.GitBranchName = headBranch is not null
+						? headBranch.Name
+						: string.Empty;
 
-				if (!_gitFetch.IsCompleted)
-				{
-					_gitFetchToken.Cancel();
-					await _gitFetch;
-					_gitFetchToken.TryReset();
+					if (!_gitFetch.IsCompleted)
+					{
+						_gitFetchToken.Cancel();
+						await _gitFetch;
+						_gitFetchToken.TryReset();
+					}
+					if (InstanceViewModel.IsGitRepository && !GitHelpers.IsExecutingGitAction)
+					{
+						_gitFetch = Task.Run(
+							() => GitHelpers.FetchOrigin(InstanceViewModel.GitRepositoryPath),
+							_gitFetchToken.Token);
+					}
 				}
-				if (InstanceViewModel.IsGitRepository && !GitHelpers.IsExecutingGitAction)
+
+				if (!GitHelpers.IsExecutingGitAction)
 				{
-					_gitFetch = Task.Run(
-						() => GitHelpers.FetchOrigin(InstanceViewModel.GitRepositoryPath),
-						_gitFetchToken.Token);
+					ContentPage.DirectoryPropertiesViewModel.UpdateGitInfo(
+						InstanceViewModel.IsGitRepository,
+						InstanceViewModel.GitRepositoryPath,
+						headBranch);
 				}
 			}
 
-			if (!GitHelpers.IsExecutingGitAction)
-			{
-				ContentPage.DirectoryPropertiesViewModel.UpdateGitInfo(
-					InstanceViewModel.IsGitRepository,
-					InstanceViewModel.GitRepositoryPath,
-					headBranch);
-			}
-
 			ContentPage.DirectoryPropertiesViewModel.DirectoryItemCount = $"{FilesystemViewModel.FilesAndFolders.Count} {directoryItemCountLocalization}";
 			ContentPage.UpdateSelectionSize();
```

Comment:
Even if a null check of ContentPage is performed at the beginning, there is a possibility that it will be nulled by other threads during processing. I think that is the cause of the exception and should be changed to `ContentPage?`.